File: src/python/validators/test_email_validator.py
import unittest
from email_validator import EmailValidator

class TestEmailValidator(unittest.TestCase):
    def test_validates_document(self):
        email_validator = EmailValidator()

        test_data = [
            ("person@example.com", True),
            ("person@co", True),
            ("name.lastname@example.com", True),
            ("person2@test.org", True),
            ("Person@Example.com", False),
            ("person+3@example.com", False),
            ("pe@rson@example.com", False),
            (".person@example.com", False),
            ("person@example.com.", False),
            ("person@", False)
        ]

        for item in test_data:
            self.assertEqual(item[1], email_validator.is_valid(item[0]))

File: src/ruby/test/test_email_validator.rb
require "minitest/autorun"
require_relative "../app/validators/email_validator.rb"

class TestEmailValidator < Minitest::Test
  def test_validates_email
    test_data = [
      { :email => "person@example.com", :expected => true },
      { :email => "person@co", :expected => true },
      { :email => "name.lastname@example.com", :expected => true },
      { :email => "person2@test.org", :expected => true },
      { :email => "Person@Example.com", :expected => false },
      { :email => "person+3@example.com", :expected => false },
      { :email => "pe@rson@example.com", :expected => false },
      { :email => ".person@example.com", :expected => false },
      { :email => "person@example.com.", :expected => false },
      { :email => "person@", :expected => false }
    ]

    test_data.each do |item|
      assert_equal(item[:expected], EmailValidator.new(item[:email]).is_valid?, "#{item[:email]} validation failed")
    end
  end
end
